spot_driver/include/spot_driver/robot_state/spot_robot_state_publisher.hpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include <spot_driver/interfaces/robot_state_client_interface.hpp>
+
+#include <spot_driver/interfaces/logger_interface_base.hpp>
+#include <spot_driver/interfaces/parameter_interface_base.hpp>
+#include <spot_driver/interfaces/tf_interface_base.hpp>
+#include <spot_driver/interfaces/timer_interface_base.hpp>
+
+#include <rclcpp/node.hpp>
```

Comment:
This dependency on rclpp is suspicious. Doesn't this invalidate the use of the middleware to sheld the publisher from the ROS infrastructure.